command.go

```diff
@@ -90,6 +90,39 @@ type Command struct {
 	// simple typo in a sub command will invoke the parent command and may
 	// end up returning a cryptic error to the user.
 	Subcommands map[string]*Command
+
+	// NoRemote denotes that a command cannot be executed in a remote environment
+	NoRemote bool
+
+	// NoLocal denotes that a command cannot be executed in a local environment
+	NoLocal bool
+
+	// Extra contains a set of other command-specific parameters
+	Extra *Extra
+}
+
+// Extra is a set of tag information for a command
+type Extra struct {
+	m map[interface{}]interface{}
```

Comment:
I'd consider just using `type Extra map[interface{}]interface{}`. That'll let users configure this using the map syntax (`Extra{SomeKnownKey: someValue}`).

Reply:
I was trying to stay away from exposing `SomeKnownKey` since thus far in our code base I've seen functions controlling the getting/setting of the `type mymarker struct{}` map entries (e.g. for the libp2p routing events).

We could take your suggestion, although we'd probably need to document the footgun of using the map directly instead of the Get method (i.e. the map may be nil).

Seem worth it to you? If so lmk and I'll make the change.

Reply:
Eh, it's fine either way.

Reply:
k, going to leave it as is and merge + bubble